include/cutlass/gemm/kernel/xe_tile_scheduler.hpp

```diff
@@ -0,0 +1,115 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ **************************************************************************************************/
+#pragma once
+
+#include "cutlass/fast_math.h"
+#include "cutlass/gemm_coord.hpp"
+#include "cutlass/kernel_hardware_info.hpp"
+#include "cute/layout.hpp"
+#include "cute/tensor.hpp"
+#include "cutlass/gemm/kernel/sm90_tile_scheduler.hpp"
+
+namespace cutlass::gemm::kernel::detail {
+
+///////////////////////////////////////////////////////////////////////////////
+
+// Persistent Thread Block (TB) scheduler
+class PersistentTileSchedulerXe : public PersistentTileSchedulerSm90 {
+
+  using BaseScheduler = PersistentTileSchedulerSm90;
+public:
+  using Params = typename BaseScheduler::Params;
+  using RasterOrder = typename BaseScheduler::RasterOrder;
+  using Arguments = typename BaseScheduler::Arguments;
+
+  CUTLASS_DEVICE explicit PersistentTileSchedulerXe(Params const& params_) : BaseScheduler(params_) {}
+
+  // Given the inputs, computes the physical grid we should launch.
+  template<class ProblemShape, class TileShape, class ClusterShape>
+  CUTLASS_HOST_DEVICE static
+  dim3
+  get_grid_shape(
+    [[maybe_unused]] Params const& params,
+    ProblemShape problem_shape,
+    TileShape tile_shape,
+    ClusterShape cluster_shape,
+    KernelHardwareInfo hw_info,
+    Arguments arguments,
+    bool truncate_by_problem_size=true) {
+
+    dim3 problem_blocks = BaseScheduler::get_tiled_cta_shape_mnl(
+      problem_shape,
+      tile_shape, cluster_shape);
+
+    int const sm_count = hw_info.sm_count;
+
+    auto problem_blocks_m = problem_blocks.x;
+    auto problem_blocks_n = problem_blocks.y;
+
+    int problem_blocks_total = problem_blocks_m * problem_blocks_n * problem_blocks.z;
+
+    RasterOrder raster_order = Params::get_rasterization_order(
+      problem_blocks_m,
+      problem_blocks_n,
+      arguments.raster_order
+    );
+
+    dim3 launch_grid;
+
+    auto possibly_truncate = [&](int x, int y) {
+      if (truncate_by_problem_size){
+        return platform::min(x, y);
+      } else {
+        return x;
+      }
+    };
+
+    // divide EUs by 8 to get available_xe_cores for PVC
+    if (raster_order == RasterOrder::AlongN) {
+        launch_grid.y = possibly_truncate(sm_count / 8, problem_blocks_total);
```

Comment:
Is 8 a magical number?  Can create a `constexpr xxx= 8`; and put a comments there what does 8 represent here as well? is it PVC specific (shall we have different value for BMG)?

Reply:
I found an easier way of avoiding this division, the [sycl_ext_oneapi_num_compute_units](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_num_compute_units.asciidoc) provides a way to query the device for the `available xe_cores` instead of the `max EUs available`.